homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -146,6 +150,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
     )
 
+    # Clean up vehicles which are not assigned to the account anymore
+    account_vehicles = {(DOMAIN, v.vin) for v in coordinator.account.vehicles}
+    device_registry = dr.async_get(hass)
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry_id=entry.entry_id
+    )
+    for device in device_entries:
+        if not device.identifiers.intersection(account_vehicles):
+            device_registry.async_remove_device(device.id)
```

Comment:
Please instead update the device entries and remove the config entry id from them. If there are no other config entries remaining the device entry will be removed automatically. That's safer than removing the device entry directly as it's (theoretically) possible for multiple config entries (from any integration) to be connected to the same device entry.

Example:
https://github.com/home-assistant/core/blob/7dc9ad63bd0503cc3298f99ae415a2718cb7f40d/homeassistant/components/tradfri/__init__.py#L191-L193

Reply:
Thanks - copied from the wrong integration :)